tests/layer_tests/tensorflow_tests/test_tf_Equal.py

```diff
@@ -0,0 +1,168 @@
+    # ir_version - common parameter
+    # use_new_frontend - common parameter
+    # specific_x_value - fills x_shape by chosen value, uses randint instead
+    # specific_y_value - fills y_shape by chosen value, uses randint instead
+    def create_tf_equal_net(self, ir_version, use_new_frontend, x_shape, y_shape, output_type, specific_x_value = None, specific_y_value = None):
+        self.x_value = specific_x_value
+        self.y_value = specific_y_value
+        self.output_type = output_type
+
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            self.x_shape = x_shape.copy() if isinstance(x_shape, list) else x_shape
+            self.y_shape = y_shape.copy() if isinstance(y_shape, list) else y_shape
+
+            if isinstance(x_shape, list):
+                self.x_shape = permute_nchw_to_nhwc(self.x_shape, use_new_frontend)
+            if isinstance(y_shape, list):
+                self.y_shape = permute_nchw_to_nhwc(self.y_shape, use_new_frontend)
+
+            if self.output_type == np.float16:
+                x = tf.compat.v1.placeholder(tf.float16, self.x_shape, 'Input')
+            elif self.output_type == np.float32:
+                x = tf.compat.v1.placeholder(tf.float32, self.x_shape, 'Input')
+            elif self.output_type == np.float64:
+                x = tf.compat.v1.placeholder(tf.float64, self.x_shape, 'Input')
+            elif self.output_type == np.int32:
+                x = tf.compat.v1.placeholder(tf.int32, self.x_shape, 'Input')
+            elif self.output_type == np.int64:
+                x = tf.compat.v1.placeholder(tf.int64, self.x_shape, 'Input')
+
+            if not self.y_value is None:
+                if isinstance(self.y_shape, np.ndarray):
+                    constant_value = np.full(self.y_shape, self.y_value, dtype=self.output_type).astype(self.output_type)
+                else:
+                    constant_value = self.output_type(self.y_value)
+            else:
+                if isinstance(self.y_shape, np.ndarray):
+                    constant_value = np.random.randint(-3, 3, self.y_shape).astype(self.output_type)
+                else:
+                    constant_value = self.output_type(self.y_shape)
+
+            y = tf.constant(constant_value)
+
+            equal = tf.equal(x, y)
+
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        ref_net = None
+        return tf_net, ref_net
+
+    test_data = [
+        # int32
+        pytest.param(
+            dict(x_shape=[2,3], y_shape=[2,3], output_type=np.int32),   #Comparing shapes with random values (verifies false and possible true)
+            marks=pytest.mark.precommit_tf_fe),
+        dict(x_shape=[2,3], y_shape=2    , output_type=np.int32),       #Comparing shape with scalar value (verifies false and possible true)
```

Comment:
`y_shape=2` is incorrect shape. If you want a vector with two elements, specify this as [2]

Reply:
moreover, these two shapes are not broadcastable

Reply:
We reviewed offline - it isn't a shape, it is a scalar value. We decided to change variable name to do not confuse.